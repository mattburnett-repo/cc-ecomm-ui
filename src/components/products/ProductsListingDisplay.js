export default function ProductListingDisplay ( props ) {
    const { data } = props;

    return (
        <div>
            <div role="presentation" aria-label="products">
                <h2>ProductsListingDisplay</h2>
                {/* Object.keys(posts).map(key => <Post key={key} body={posts[key]} />)  */}
                {data.map((item, index) => (
                    <div role="presentation" aria-label='product' key={index}>
                        <label htmlFor='product-name'>Product Name:</label>
                        <div role='presentation' aria-label='product-name'>{item.name}</div>
                        <label htmlFor='product-description'>Product Description:</label>
                        <div role='presentation' aria-label='product-description'>{item.description}</div>
                        <label htmlFor='product-image-url'>Image:</label>
                        <div role="img" id="product-image-url" aria-label='product-image-url'>
                            <img src="item.image_url" alt={item.name} />
                        </div>
                        <label htmlFor='product-price'>Product Price:</label>
                        <div role='presentation' aria-label='product-price'>{item.price}</div>
                    </div>
                ))}     
            </div>
        </div>
    )
}