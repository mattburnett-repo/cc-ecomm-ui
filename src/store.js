import { configureStore } from '@reduxjs/toolkit';

import authReducer from './features/auth/authSlice';
import productCategoryReducer from './features/productCategory/productCategorySlice'
import productsListingReducer from './features/products/productsSlice'
import ordersReducer from './features/orders/ordersSlice'
import cartsReducer from './features/carts/cartsSlice'

function rootReducer() {
    return {
        reducer: {
            auth: authReducer,
            productCategories: productCategoryReducer,
            orders: ordersReducer,
            carts: cartsReducer,
            products: productsListingReducer,
        }    
    }
}

const store = configureStore(rootReducer());

export default store;