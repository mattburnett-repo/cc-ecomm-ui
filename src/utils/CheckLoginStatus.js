
import { useSelector } from 'react-redux'
import { useHistory } from 'react-router-dom'

import { selectIsAuthenticated } from '../features/auth/authSlice'

export default function CheckLoginStatus () {
    // eslint-disable-next-line
    const isLoggedIn = useSelector(selectIsAuthenticated);

    const history = useHistory()

    if(!isLoggedIn) {
        history.push('/login')

        return false
    } else {
        return true;
    }
}        
