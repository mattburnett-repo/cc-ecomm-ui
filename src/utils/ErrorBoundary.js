import React from 'react'

export default class ErrorBoundary extends React.Component {
    constructor(props) {
      super(props);
      this.state = { hasError: false };
    }
  
    static getDerivedStateFromError(error) {
      // Update state to trigger fallback UI
      return { hasError: true };
    }
  
    componentDidCatch(error, errorInfo) {
        // Log error or send logging data to log management tool
        //   logErrorToMyService(error, errorInfo);
        console.log('ErrorBoundary ', errorInfo)
    }
  
    render() {
      if (this.state.hasError) {
        // Render fallback UI
        return <h1>There was an error. Sorry about that; we're working on it.</h1>;
      }
  
      return this.props.children; 
    }
  }

