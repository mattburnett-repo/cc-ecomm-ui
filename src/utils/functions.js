
// USAGE: formatAsCurrency.format(3.005); // "3.01"
const formatAsCurrency = new Intl.NumberFormat('en-US', {
    minimumFractionDigits: 2,
    maximumFractionDigits: 2,
});

const formatDate =   new Intl.DateTimeFormat("en-US", {
    year: "numeric",
    month: "long",
    day: "2-digit"
});

module.exports = {
    formatAsCurrency: formatAsCurrency,
    formatDate: formatDate
}