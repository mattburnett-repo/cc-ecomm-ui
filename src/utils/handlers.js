
// handler function declarations
function handleImageClick (e) {
    alert('handleImageClick ' + e.target.id)
    // redirect to product-detail with e.target.id as productId
}

function handleAddToCart(e){ 
    alert('handleAddToCart inside utils product_id: ' + e.target.id + ' quantity: ' + e.quantity)
} 

function handleCheckout(e){ 
    alert('handleCheckout inside utils ')
} 

function handleSearch(e){
    e.preventDefault()
    // let searchTerms = e.target.value
    alert('handleSearch for search term/s ' + e.target.searchTerms.value + '. \nneed to figure out how to get dispatch() to work from here')
}

function handleOrderClick(e) {
    e.preventDefault()

    let order_id = e.target.order_id.value
    alert('go to order detail component ' + order_id)
}

function handleHomeClick() {
    console.log('handleHomeClick')
}
function handleShowCartClick() {
    console.log('handleShowCartClick')
}
function handleCheckoutClick() {
    console.log('handleCheckoutClick')
}
function handleCancelClick() {
    console.log('handleCanceClick')
}


module.exports = {
    handleHomeClick: handleHomeClick,
    handleShowCartClick: handleShowCartClick,
    handleCheckoutClick: handleCheckoutClick,
    handleCancelClick: handleCancelClick,
    handleImageClick: handleImageClick,
    handleAddToCart: handleAddToCart,
    handleCheckout: handleCheckout,
    handleSearch: handleSearch,
    handleOrderClick: handleOrderClick
} // end module.exports