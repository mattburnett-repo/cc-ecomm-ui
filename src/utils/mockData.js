// https://css-tricks.com/writing-tests-for-react-applications-using-jest-and-enzyme/

module.exports = {
    // someday I will figure out how to do this correctly...
    // getUserData: jest.fn(() => {
    //   return Promise.resolve({
    //   data: [
    //     {
    //       id: 1,
    //       username: 'asdf',
    //       email: 'asdf@asdf.com',
    //       password: 'asdf'
    //     }
    //   ]
    //   })
    // }),
    mockCartsData:[
        {
          "cart": {
              "cart_id": 1,
              "user_id": 1,
              "cart_items": [
                  {
                      "product_id": 1,
                      "product_name": "hedgehog",
                      "product_price": 1,
                      "product_quantity": 1,
                      "product_total_price": 1
                  },
                  {
                      "product_id": 2,
                      "product_name": "maine coon cat",
                      "product_price": 2,
                      "product_quantity": 2,
                      "product_total_price": 4
                  },
                  {
                      "product_id": 3,
                      "product_name": "bulldog",
                      "product_price": 3,
                      "product_quantity": 3,
                      "product_total_price": 9
                  },
                  {
                      "product_id": 4,
                      "product_name": "bunny",
                      "product_price": 4,
                      "product_quantity": 4,
                      "product_total_price": 16
                  }
              ],
              "cart_total_price": {
                  "sum": 30,
                  "cart_id": 1
              }
          }
      }
    ],
    mockCartData: {
      cart: {
        cart_id: 1,
        user_id: 1,
        cart_items: [
          {
            product_id: 1,
            product_name: 'hedgehog',
            product_price: 1,
            product_quantity: 1,
            product_total_price: 1
          },
          {
            product_id: 2,
            product_name: 'maine coon cat',
            product_price: 2,
            product_quantity: 2,
            product_total_price: 4
          },
          {
            product_id: 3,
            product_name: 'bulldog',
            product_price: 3,
            product_quantity: 3,
            product_total_price: 9
          },
          {
            product_id: 4,
            product_name: 'bunny',
            product_price: 4,
            product_quantity: 4,
            product_total_price: 16
          }
      ],
      cart_total_price: { sum: 30, cart_id: 1 }
    }},
    mockOrderData: [{
      "order": {
          "cart": {
              "cart_id": 1,
              "cart_items": [
                  {
                      "product_id": 1,
                      "product_name": "product_name_01",
                      "product_price": 1,
                      "product_quantity": 1,
                      "product_total_price": 1
                  },
                  {
                      "product_id": 2,
                      "product_name": "product_name_02",
                      "product_price": 2,
                      "product_quantity": 2,
                      "product_total_price": 4
                  },
                  {
                      "product_id": 3,
                      "product_name": "product_name_03",
                      "product_price": 3,
                      "product_quantity": 3,
                      "product_total_price": 9
                  },
                  {
                      "product_id": 4,
                      "product_name": "product_name_04",
                      "product_price": 4,
                      "product_quantity": 4,
                      "product_total_price": 16
                  }
              ]
          },
          "user_id": 1,
          "order_id": 1,
          "order_date": null,
          "total_price": {
              "sum": 30,
              "cart_id": 1
          }
        }
      }],
    mockOrderConfirmationData:
      {
        "order_id": 1,
        "total_amount": 1234,
        "order_date": '12-Dez-2021',
        "payment_type": 'Deutschmarks',
        "delivery_address": '123 Katzendamm, 90210 Berlin DE'
      },
    mockProductCategories: [
        // {
        //   "category_id": 0,
        //   "description": "Please select a product category"
        // },
        {
          "category_id": 4,
          "description": "Auto"
        },
        {
            "category_id": 3,
            "description": "Health"
        },
        {
            "category_id": 1,
            "description": "Pets"
        },
        {
            "category_id": 2,
            "description": "Tech"
        }
      ],
    mockProduct:[
        {
          "id": 2,
          "category_id": 2,
          "name": "product_name_02",
          "description": "product_desc_02",
          "image_url": "https://cdn.pixabay.com/photo/2015/03/27/13/16/maine-coon-694730_1280.jpg",
          "price": "2.00"
        },      
      ],
    mockProducts: [
        {
            "id": 1,
            "category_id": 1,
            "name": "product_name_01",
            "description": "product_desc_01",
            "image_url": "https://pixabay.com/images/id-71282/",
            "price": "1.00"
        },
        {
            "id": 2,
            "category_id": 2,
            "name": "product_name_02",
            "description": "product_desc_02",
            "image_url": "http://www.image.com/item/2",
            "price": "2.00"
        },
        {
            "id": 3,
            "category_id": 3,
            "name": "product_name_03",
            "description": "product_desc_03",
            "image_url": "http://www.image.com/item/3",
            "price": "3.00"
        },
        {
            "id": 4,
            "category_id": 4,
            "name": "product_name_04",
            "description": "product_desc_04",
            "image_url": "http://www.image.com/item/4",
            "price": "4.00"
        }
    ],
    mockUserData:[
      {
        "id": 4,
        "user_name": "asdf",
        "password": "$2b$10$OsTX1L59rYgsmAnsrJsXoO3RH3RSfLgEF53YBB3723B9eoELtUa.a",
        "email": "asdf@asdf.com",
        "google_id": null,
        "google_display_name": null,
        "google_first_name": null,
        "google_last_name": null,
        "google_image": null
      }
    ]
  } // end module.exports