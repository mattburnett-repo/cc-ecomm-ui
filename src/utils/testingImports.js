// export default function TestingImports() {
//     return (
//         import { render, screen, fireEvent } from '@testing-library/react';
//         import renderer from 'react-test-renderer';
//         import userEvent from '@testing-library/user-event';

//         import { Provider } from 'react-redux';
//         import store from '../../store'
//     )
// }