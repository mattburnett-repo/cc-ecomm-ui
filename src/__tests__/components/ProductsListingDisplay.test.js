// 20211217: Wireframe https://wireframe.cc/D5VIOi

import { render, screen, fireEvent, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
// import userEvent from '@testing-library/user-event';

import { Provider } from 'react-redux';
import store from '../../store'

import ProductsListingDisplay from '../../components/products/ProductsListingDisplay';

import { mockProducts } from '../../utils/mockData'

const mockHandleImageClick = jest.fn()

function handlers() {
    return {
        handleImageClick: mockHandleImageClick
    }
}

describe('<ProductsListingDisplay data={mockData} /> component tests', () => {
    beforeEach(() => render(<Provider store={store}><ProductsListingDisplay data={mockProducts} handlers={handlers()}/></ Provider>));

    it('should display products', () => { 
        screen.getByRole('presentation', {name: /products/i});
        
        let theVals = screen.getAllByRole('presentation', {name: /^product$/i});
        expect(theVals).toHaveLength(4);

        expect(theVals[0]).toHaveTextContent('product_name_01');
        expect(theVals[1]).toHaveTextContent('product_name_02');
        expect(theVals[2]).toHaveTextContent('product_name_03');
        expect(theVals[3]).toHaveTextContent('product_name_04');
    });
    it('should render product names', () => { 
        let theVal = screen.getAllByRole('presentation', {name: /product-name/i});
        expect(theVal).toHaveLength(4);
    });
    it('should render product descriptions', () => {
        let theVal = screen.getAllByRole('presentation', {name: /product-description/i});
        expect(theVal).toHaveLength(4);
    });
    it('should render product image_urls', () => { 
        let theVal = screen.getAllByRole('img', {name: /image-url/i});
        expect(theVal).toHaveLength(4);
    });
    it('should click the image as a link', () => {
        let theVals = screen.getAllByRole('img', /product-image-url/i)
        fireEvent.click(theVals[0])

    })
    it('should render product prices', () => { 
        let theVal = screen.getAllByRole('presentation', {name: /product-price/i});
        expect(theVal).toHaveLength(4);
    });

    it('creates a snapshot', () => {
        const tree = renderer.create(<Provider store={store}><ProductsListingDisplay data={mockProducts}  handlers={handlers()} /></ Provider>).toJSON();
        expect(tree).toMatchSnapshot();  
    });
}); // end component
