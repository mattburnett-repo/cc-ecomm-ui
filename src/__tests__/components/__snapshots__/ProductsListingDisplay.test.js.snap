// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductsListingDisplay data={mockData} /> component tests creates a snapshot 1`] = `
<div>
  <div
    aria-label="products"
    role="presentation"
  >
    <h2>
      ProductsListingDisplay
    </h2>
    <div
      aria-label="product"
      role="presentation"
    >
      <label
        htmlFor="product-id"
      >
        Product ID:
      </label>
      <div
        aria-label="product-id"
        role="presentation"
      >
        1
      </div>
      <label
        htmlFor="product-name"
      >
        Product Name:
      </label>
      <div
        aria-label="product-name"
        role="presentation"
      >
        product_name_01
      </div>
      <label
        htmlFor="product-description"
      >
        Product Description:
      </label>
      <div
        aria-label="product-description"
        role="presentation"
      >
        product_desc_01
      </div>
      <label
        htmlFor="product-image-url"
      >
        Image:
      </label>
      <div
        aria-label="product-image-url"
        id="product-image-url"
        role="img"
      >
        <img
          alt="product_name_01"
          id={1}
          onClick={[MockFunction]}
          src="https://pixabay.com/images/id-71282/"
        />
      </div>
      <label
        htmlFor="product-price"
      >
        Product Price:
      </label>
      <div
        aria-label="product-price"
        role="presentation"
      >
        1.00
      </div>
      <label
        htmlFor={1}
      >
        Quantity:
      </label>
      <input
        aria-label="quantity"
        id={1}
        name="quantity"
        placeholder="Enter product quantity"
      />
      <button
        aria-label="add-to-cart"
        id={1}
        onClick={[MockFunction]}
      >
        Add To Cart
      </button>
    </div>
    <div
      aria-label="product"
      role="presentation"
    >
      <label
        htmlFor="product-id"
      >
        Product ID:
      </label>
      <div
        aria-label="product-id"
        role="presentation"
      >
        2
      </div>
      <label
        htmlFor="product-name"
      >
        Product Name:
      </label>
      <div
        aria-label="product-name"
        role="presentation"
      >
        product_name_02
      </div>
      <label
        htmlFor="product-description"
      >
        Product Description:
      </label>
      <div
        aria-label="product-description"
        role="presentation"
      >
        product_desc_02
      </div>
      <label
        htmlFor="product-image-url"
      >
        Image:
      </label>
      <div
        aria-label="product-image-url"
        id="product-image-url"
        role="img"
      >
        <img
          alt="product_name_02"
          id={2}
          onClick={[MockFunction]}
          src="http://www.image.com/item/2"
        />
      </div>
      <label
        htmlFor="product-price"
      >
        Product Price:
      </label>
      <div
        aria-label="product-price"
        role="presentation"
      >
        2.00
      </div>
      <label
        htmlFor={2}
      >
        Quantity:
      </label>
      <input
        aria-label="quantity"
        id={2}
        name="quantity"
        placeholder="Enter product quantity"
      />
      <button
        aria-label="add-to-cart"
        id={2}
        onClick={[MockFunction]}
      >
        Add To Cart
      </button>
    </div>
    <div
      aria-label="product"
      role="presentation"
    >
      <label
        htmlFor="product-id"
      >
        Product ID:
      </label>
      <div
        aria-label="product-id"
        role="presentation"
      >
        3
      </div>
      <label
        htmlFor="product-name"
      >
        Product Name:
      </label>
      <div
        aria-label="product-name"
        role="presentation"
      >
        product_name_03
      </div>
      <label
        htmlFor="product-description"
      >
        Product Description:
      </label>
      <div
        aria-label="product-description"
        role="presentation"
      >
        product_desc_03
      </div>
      <label
        htmlFor="product-image-url"
      >
        Image:
      </label>
      <div
        aria-label="product-image-url"
        id="product-image-url"
        role="img"
      >
        <img
          alt="product_name_03"
          id={3}
          onClick={[MockFunction]}
          src="http://www.image.com/item/3"
        />
      </div>
      <label
        htmlFor="product-price"
      >
        Product Price:
      </label>
      <div
        aria-label="product-price"
        role="presentation"
      >
        3.00
      </div>
      <label
        htmlFor={3}
      >
        Quantity:
      </label>
      <input
        aria-label="quantity"
        id={3}
        name="quantity"
        placeholder="Enter product quantity"
      />
      <button
        aria-label="add-to-cart"
        id={3}
        onClick={[MockFunction]}
      >
        Add To Cart
      </button>
    </div>
    <div
      aria-label="product"
      role="presentation"
    >
      <label
        htmlFor="product-id"
      >
        Product ID:
      </label>
      <div
        aria-label="product-id"
        role="presentation"
      >
        4
      </div>
      <label
        htmlFor="product-name"
      >
        Product Name:
      </label>
      <div
        aria-label="product-name"
        role="presentation"
      >
        product_name_04
      </div>
      <label
        htmlFor="product-description"
      >
        Product Description:
      </label>
      <div
        aria-label="product-description"
        role="presentation"
      >
        product_desc_04
      </div>
      <label
        htmlFor="product-image-url"
      >
        Image:
      </label>
      <div
        aria-label="product-image-url"
        id="product-image-url"
        role="img"
      >
        <img
          alt="product_name_04"
          id={4}
          onClick={[MockFunction]}
          src="http://www.image.com/item/4"
        />
      </div>
      <label
        htmlFor="product-price"
      >
        Product Price:
      </label>
      <div
        aria-label="product-price"
        role="presentation"
      >
        4.00
      </div>
      <label
        htmlFor={4}
      >
        Quantity:
      </label>
      <input
        aria-label="quantity"
        id={4}
        name="quantity"
        placeholder="Enter product quantity"
      />
      <button
        aria-label="add-to-cart"
        id={4}
        onClick={[MockFunction]}
      >
        Add To Cart
      </button>
    </div>
  </div>
</div>
`;
